### PR TITLE
Add machine-readable compatibility matrix and Gateway API compat info

### LINKED ISSUE #4007: Include Compatibility Matrix info as part of Contour Release
Today we maintain version compatibility between Contour and Envoy here https://projectcontour.io/resources/compatibility-matrix/  

To better align with the helm chart maintained by Bitnami team, we should include this info as a JSON as part of our release artifacts each time (see https://github.com/projectcontour/contour/issues/3943) .  This is much easier than parsing html.  Bitnami will leverage this in their verification tests to automatically ensure the prescribed Contour and Envoy versions in the helm chart are indeed compatible.

### PR DESCRIPTION
**Pull request by youngnick**
This adds a machine readable file that records the same information as
the "Compatibility Matrix" page.

It also adds Gateway API version info to both places.

Closes #4007.

Signed-off-by: Nick Young <ynick@vmware.com>

### CODE REVIEW

File: compatibility.yaml [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  repository: "https://github.com/projectcontour/contour"
+versions:
+  - version: "main"
+    dependencies:
+      envoy: "1.19.1"
+      kubernetes:
+        - "1.22"
+        - "1.21"
+        - "1.20"
+      gateway-api: v1alpha1
```
Comment (stevesloka) [inline compatibility.yaml:14]: Maybe make this a list so that we can support multiple versions at once?

**Review by stevesloka — CHANGES_REQUESTED** [review]

**Review by sunjayBhatia — COMMENTED** [review]
should we add some linting check to make sure this stays valid yaml and doesnt break anyone consuming it?

**Comment by youngnick** [comment]
That's a great idea @sunjayBhatia, I'll see what I can do.

**Comment by youngnick** [comment]
Okay, I think that's all the changes. PTAL.

**Comment by youngnick** [comment]
I also added another "supported" field - I realized this would be nice when I was writing the docs for the quarterly release process.

File: site/content/resources/compatibility-matrix.md [hunk 2]
```diff
@@ -8,29 +8,29 @@ These combinations of versions are specifically tested in CI and supported by th
...
-| 1.13.1          | [1.17.1][6]          | 1.20, 1.19, 1.18    | [1.13.1][54]     |
-| 1.13.0          | [1.17.0][5]          | 1.20, 1.19, 1.18    | [1.13.0][53]     |
-| 1.12.0          | [1.17.0][5]          | 1.19, 1.18, 1.17    | [1.12.0][52]     |
-| 1.11.0          | [1.16.2][4]          | 1.19, 1.18, 1.17    | [1.11.0][51]     |
-| 1.10.1          | [1.16.2][4]          | 1.19, 1.18, 1.17    | N/A              |
-| 1.10.0          | [1.16.0][3]          | 1.19, 1.18, 1.17    | N/A              |
-| 1.9.0           | [1.15.1][2]          | 1.18, 1.17, 1.16    | N/A              |
+| Contour Version | Envoy Version        | Kubernetes Versions | Operator Version | Gateway API Version |
+| --------------- | :------------------- | ------------------- | ---------------- | --------------------|
+| main            | [1.19.1][17]         | 1.22, 1.21, 1.20    | [main][50]       | v1alpha1            |
```
Comment (skriss) [inline site/content/resources/compatibility-matrix.md:13]: Looks like you may have run into some merge conflict issues with the Envoy version link numbers here -- the ones you have are no longer valid, need to switch to what's in `main`.

**Review by skriss — CHANGES_REQUESTED** [review]

File: site/content/resources/compatibility-matrix.md (hunk 2 above)
Comment (youngnick) [reply site/content/resources/compatibility-matrix.md:13]: dang, thanks, I'll update.

File: site/content/resources/compatibility-matrix.md [hunk 3]
```diff
@@ -8,29 +8,29 @@ These combinations of versions are specifically tested in CI and supported by th
...
+| 1.18.0          | [1.19.0][10]         | 1.21, 1.20, 1.19    | [1.18.0][61]     | v1alpha1            |
+| 1.17.2          | [1.18.4][12]         | 1.21, 1.20, 1.19    | N/A              | v1alpha1            |
+| 1.17.1          | [1.18.3][9]         | 1.21, 1.20, 1.19    | N/A              | v1alpha1            |
+| 1.17.0          | [1.18.3][9]         | 1.21, 1.20, 1.19    | [1.17.0][60]     | v1alpha1            |
+| 1.16.1          | [1.18.4][12]         | 1.21, 1.20, 1.19    | N/A              | v1alpha1            |
+| 1.16.0          | [1.18.3][9]         | 1.21, 1.20, 1.19    | [1.16.0][59]     | v1alpha1            |
+| 1.15.2          | [1.18.4][12]         | 1.21, 1.20, 1.19    | N/A              | v1alpha1            |
+| 1.15.1          | [1.18.3][9]         | 1.21, 1.20, 1.19    | [1.15.1][58]     | v1alpha1            |
+| 1.15.0          | [1.18.2][8]         | 1.21, 1.20, 1.19    | [1.15.0][57]     | v1alpha1            |
+| 1.14.2          | [1.17.4][7]         | 1.20, 1.19, 1.18    | N/A              | v1alpha1            |
```
Comment (skriss) [inline site/content/resources/compatibility-matrix.md:24]:
```suggestion
| 1.14.2          | [1.17.4][11]         | 1.20, 1.19, 1.18    | N/A              | v1alpha1            |
```

File: site/content/resources/compatibility-matrix.md [hunk 4]
```diff
@@ -8,29 +8,29 @@ These combinations of versions are specifically tested in CI and supported by th
...
+| 1.14.1          | [1.17.2][7]         | 1.20, 1.19, 1.18    | [1.14.1][56]     | v1alpha1            |
+| 1.14.0          | [1.17.1][6]         | 1.20, 1.19, 1.18    | [1.14.0][55]     | v1alpha1            |
+| 1.13.1          | [1.17.1][6]         | 1.20, 1.19, 1.18    | [1.13.1][54]     | v1alpha1            |
+| 1.13.0          | [1.17.0][5]          | 1.20, 1.19, 1.18    | [1.13.0][53]     | v1alpha1            |
+| 1.12.0          | [1.17.0][5]          | 1.19, 1.18, 1.17    | [1.12.0][52]     | N/A                 |
+| 1.11.0          | [1.16.2][4]          | 1.19, 1.18, 1.17    | [1.11.0][51]     | N/A                 |
+| 1.10.1          | [1.16.2][4]          | 1.19, 1.18, 1.17    | N/A              | N/A                 |
+| 1.10.0          | [1.16.0][2]          | 1.19, 1.18, 1.17    | N/A              | N/A                 |
```
Comment (skriss) [inline site/content/resources/compatibility-matrix.md:32]:
```suggestion
| 1.10.0          | [1.16.0][3]          | 1.19, 1.18, 1.17    | N/A              | N/A                 |
```

File: site/content/resources/compatibility-matrix.md [hunk 5]
```diff
@@ -8,29 +8,29 @@ These combinations of versions are specifically tested in CI and supported by th
...
+| 1.9.0           | [1.15.1][1]          | 1.18, 1.17, 1.16    | N/A              | N/A                 |
```
Comment (skriss) [inline site/content/resources/compatibility-matrix.md:33]:
```suggestion
| 1.9.0           | [1.15.1][2]          | 1.18, 1.17, 1.16    | N/A              | N/A                 |
```

**Comment by youngnick** [comment]
Thanks again @skriss, I checked them properly this time. 🤦

File: site/content/resources/release-process.md [hunk 6]
```diff
@@ -49,6 +49,7 @@ go run ./hack/release/prepare-release.go $CONTOUR_RELEASE_VERSION
 ```
 
 1. Add the new release to the compatibility matrix (`site/content/resources/compatibility-matrix.md`).
+1. Add the new release to the compatibility YAML (`/compatibility.yaml`).
```
Comment (sunjayBhatia) [inline site/content/resources/release-process.md:52]: versions.yaml ?

Comment (youngnick) [reply site/content/resources/release-process.md:52]: 🤦

**Review by skriss — APPROVED** [review]

**Review by stevesloka — APPROVED** [review]

**Pull request merged**